R/zchunk_batch_building_SSP.xml.R

```diff
@@ -0,0 +1,85 @@
+#' module_energy_batch_building_SSP.xml
+#'
+#' Construct XML data structure for all the \code{building_SSP.xml} files.
+#'
+#' @param command API command to execute
+#' @param ... other optional parameters, depending on command
+#' @return Depends on \code{command}: either a vector of required inputs,
+#' a vector of output names, or (if \code{command} is "MAKE") all
+#' the generated outputs: \code{building_SSP1.xml}, \code{building_SSP2.xml}, \code{building_SSP3.xml},
+#' \code{building_SSP4.xml},and \code{building_SSP5.xml}.
+module_energy_batch_building_SSP.xml <- function(command, ...) {
+  if(command == driver.DECLARE_INPUTS) {
+    return(c(paste0("L244.Satiation_flsp_SSP", SSP_NUMS),
+             paste0("L244.SatiationAdder_SSP", SSP_NUMS),
+             paste0("L244.GenericServiceSatiation_SSP", SSP_NUMS),
+             paste0("L244.FuelPrefElast_bld_SSP", unique(fuelpref_NUMS)),
+             "L244.DeleteThermalService"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c(XML = "building_SSP1.xml",
+             XML = "building_SSP2.xml",
+             XML = "building_SSP3.xml",
+             XML = "building_SSP4.xml",
+             XML = "building_SSP5.xml"))
+  } else if(command == driver.MAKE) {
+
+    # silence package checks
+    building_SSP1.xml <- building_SSP2.xml <- building_SSP3.xml <- building_SSP4.xml <-
+      building_SSP5.xml <- NULL
+
+    SSP_NUMS <- seq(1,5)
+    fuelpref_NUMS <- c("1" = "15", "3" = "3", "4" = "4", "5" = "15")
+
+    all_data <- list(...)[[1]]
+
+    for (i in SSP_NUMS){
+      # Names of inputs and outputs
+      Satiation_flsp <- paste0("L244.Satiation_flsp_SSP", i)
+      SatiationAdder <- paste0("L244.SatiationAdder_SSP", i)
+      GenericServiceSatiation <- paste0("L244.GenericServiceSatiation_SSP", i)
+      # SSP2 is unique, doesn't have fuelprefelast or deletethermalservice
+      if (i != 2){
```

Comment:
?

Reply:
SSP2 uses less files than the other SSPs (because they are all changing from the default)